### ISSUE TITLE
A rule options object changes unrelated alt-text and no-invalid-void-type defaults

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1223]
## Summary

With @utoo/lint 0.4.1, adding an options object to prefer-promise-reject-errors changes the diagnostics of two unrelated rules. This is reproducible with only three public rules, no plugins, no imports, and no disable comments.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1
```

`example.tsx`:

```tsx
export function Example() { return <img src="image.png" />; }
export interface Options { run: () => Promise<void>; }
```

`base.json`:

```json
{
  "rules": {
    "jsx-a11y/alt-text": "warn",
    "@typescript-eslint/no-invalid-void-type": "error",
    "prefer-promise-reject-errors": "warn"
  }
}
```

`with-options.json`:

```json
{
  "rules": {
    "jsx-a11y/alt-text": "warn",
    "@typescript-eslint/no-invalid-void-type": "error",
    "prefer-promise-reject-errors": [
      "warn",
      {
        "allowEmptyReject": true
      }
    ]
  }
}
```

```sh
./node_modules/.bin/utlint --config base.json --format=json example.tsx
./node_modules/.bin/utlint --config with-options.json --format=json example.tsx
```

## Actual

- With base.json: one jsx-a11y/alt-text warning, no no-invalid-void-type error.
- With with-options.json: the alt-text warning disappears, and Promise<void> gets an @typescript-eslint/no-invalid-void-type error.
- Changing allowEmptyReject to false also produces the second result.

## Expected

Both configurations should produce the first result. The sample has no Promise.reject call, so this options change should have no effect. It must not disable image alt checks or change whether void is allowed as a generic type argument.

Verified on macOS arm64 with Node.js 22.23.1.
